pandas/_libs/lib.pyx

```diff
@@ -1299,12 +1323,18 @@ cdef class Seen:
         return (self.uint_ and (self.null_ or self.sint_)
                 and not self.coerce_numeric)
 
-    cdef saw_null(self):
+    cdef saw_null(self, seen_float=True):
         """
         Set flags indicating that a null value was encountered.
+
+        Parameters
+        ----------
+        seen_float: boolean, default True
```

Comment:
The docstring here is a little unclear; the keyword is `seen_float` but this makes it seem like it should be called `seen_nan`, though that's also a bit confusion since the entire function is `saw_null`.

Reply:
``seen_float`` is there to not break ``maybe_convert_objects``. For some reason, ``seen_null`` today means ``seen_nan``(floating NaN)

In the future, I plan on getting rid of this parameter. (It's probably better to go through ``seen.set_curr_val`` for floating NaNs)

Reply:
agreed it would be nice to make the naming/docs unambiguous.  I'm a big fan of breaking things like this off into precursor PRs